k-distribution/tests/regression-new/itp/nth-ancestor/nth1-spec.k

```diff
@@ -23,8 +23,6 @@
     by apply/connect1.
 */
 
```

Comment:
That's what I did initially (see my first commit).
But our testing infrastructure is not smart enough to remove paths from the include directory, and Jenkins started to fail.
```
[Error] Outer Parser: Module LIST previously declared at Source(/home/radu/work/k/k-distribution/target/release/k/include/kframework/builtin/domains.md) and Location(631,1,746,9)
```
To add a test that exercises the reported error would involve significant changes to the makefiles, and I don't think they are warranted just for this test.

Reply:
I thought that a script removed some of these absolute paths and replaced them with just the filenames before comparing against a saved diff

Reply:
I just checked the test suite. When you're running something like `make -j8 -O -k` from the `regression-new/checks` directory, I see this command:
`kompile -w none --no-exc-wrap --backend llvm  badFunctionRuleWithContext.k -d . 2>&1 | sed 's!'`pwd`'/\(\./\)\{0,2\}!!g' | diff - badFunctionRuleWithContext.k.out`
but this removes the current directory. Maybe you could add a test that will fail from a duplicate module that's defined in the test suite's working directory rather than `domains.md`. You could even duplicate definition of the same module in the same file.

Reply:
We already have such a test: https://github.com/kframework/k/blob/master/k-distribution/tests/regression-new/checks/duplicateModule.k
Again, check the first commit in this PR.
The problem was that we had separate checks for domains.md vs user definition.
To fix this I put together the previousModules with the current modules and run the check again.